solr/core/src/java/org/apache/solr/cluster/placement/plugins/OrderedNodePlacementPlugin.java

```diff
@@ -515,6 +521,7 @@ public final void removeReplica(Replica replica) {
           });
       if (hasReplica.get()) {
         removeProjectedReplicaWeights(replica);
+        allReplicas.remove(replica);
```

Comment:
Now that we're separately tracking replicas in a top-level map (and assuming that this is all accessed by a single thread), I think this method could be simplified to:
```java
if (allReplicas.remove(replica)) {
  // [the nested removal logic]
}
```

Reply:
Good catch @magibney ! I'm wondering if these 2 statements should just placed in https://github.com/apache/solr/blob/d8902468b8b292cb37ee3e1ba8d1a724752ab9a8/solr/core/src/java/org/apache/solr/cluster/placement/plugins/OrderedNodePlacementPlugin.java#L516 directly?

Perhaps I have overlooked some edge cases which using the AtomicBoolean hasReplica was necessary ?

Reply:
After i read the code again, I'm not 100% sure about the `hasReplica` flag. 

However, it does make the logic a bit more isolated and easier to follow - iterate the list first, figure out if the replica exists, if it does, deal with it AFTEr finishing the traversal. In a logical level, it's pretty clean and avoid any concurrent modification exception (though the map/list being iterated on is private field anyway).

Hm...I simply move the allReplicas.remove to within the `if (reps.remove(replica))` block now, cause i do think it's the best if the operations (add/remove) on the 2 collections stay close to each other.

Anyway, I think these are very minor concerns? 😊 either ways are fine